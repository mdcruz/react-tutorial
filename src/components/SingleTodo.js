import React from 'react';

export default class SingleTodo extends React.Component {
    render () {
        return (
            <li className={this.props.isDone ? "done": ""}>
                <input
                    checked={this.props.isDone}
                    onChange={() => this.props.completeTodo(this.props.todoId)}
                    type="checkbox"/>
                <label>{this.props.text}</label>
                <button id="deleteButton"
                    onClick={() => this.props.removeTodo(this.props.todoId)}>
                        Delete
                </button>
            </li>
        );
    }
}