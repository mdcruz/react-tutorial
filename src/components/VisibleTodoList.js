import React from 'react';
import SingleTodo from './SingleTodo'

export default class VisibleTodoList extends React.Component {
    render() {
        return (
            <div>
                <h3>{this.props.visibilityFilter.replace('_', ' ')}</h3>
                {this.props.visibleTodos.length > 0 ?
                    (
                        <ul>
                            {this.props.visibleTodos.map(
                                (todo) =>
                                    <SingleTodo
                                        key={todo.id}
                                        todoId={todo.id}
                                        text={todo.descriptionText}
                                        isDone={todo.isDone}
                                        completeTodo={this.props.completeTodo}
                                        removeTodo={this.props.removeTodo}
                                    />
                            )}
                        </ul>
                    ):
                    (
                        <ul>
                            <li>Nothing to see here</li>
                        </ul>
                    )
                }
            </div>
        );
    }
}