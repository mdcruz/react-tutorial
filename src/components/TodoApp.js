import React from 'react';
import VisibleTodoList from './VisibleTodoList';

export default class TodoApp extends React.Component {
    constructor(props) {
        super(props);
        this.visibilityFilters = ["ALL_TODOS", "ACTIVE_TODOS", "COMPLETED_TODOS"];
        this.state = {
            todos: this.props.todoActions.getAllTodos(),
            visibilityFilter: 'ALL_TODOS',
            value: ''
        };
    }

    handleChange = (event) => {
      this.setState({value: event.target.value});
    }
  
    addTodo = (event) => {
      this.props.todoActions.addTodo(this.state.value);
      this.setState({todos: this.props.todoActions.getAllTodos()});
      this.setState({ value: '' }); 
      event.preventDefault();
    }

    completeTodo = (todoId) => {
        this.props.todoActions.completeTodo(todoId);
        this.setState({todos: this.props.todoActions.getAllTodos()});
    }

    removeTodo = (todoId) => {
        this.props.todoActions.removeTodo(todoId);
        this.setState({todos: this.props.todoActions.getAllTodos()});
    }

    changeVisibilityFilter = (visibilityFilter) => {
        this.setState({ visibilityFilter: visibilityFilter });
    }

    getTodos = () => {
        switch (this.state.visibilityFilter) {
            case "ALL_TODOS":
                return this.state.todos;
            case "ACTIVE_TODOS":
                return this.state.todos.filter(todo => todo.isDone === false);
            case "COMPLETED_TODOS":
                return this.state.todos.filter(todo => todo.isDone === true);
            default:
                return this.state.todos;
        }
    }
    
    render() {
        return (
            <div>
                <h2 className="appTitle">Minimal Todo App built with React</h2>
                <form className="addTodoForm" onSubmit={(e) => this.addTodo(e)}>
                    <input className="todoField" type="text" value={this.state.value} onChange={(e) => this.handleChange(e)} placeholder="What do you want to do today?" />
                    <button disabled={!this.state.value}>Add todo</button>
                </form>
                <VisibleTodoList
                    visibleTodos={this.getTodos()}
                    visibilityFilter = {this.state.visibilityFilter}
                    completeTodo={this.completeTodo}
                    removeTodo={this.removeTodo}
                />
                <div> {
                    this.visibilityFilters.map(
                        visibilityFilter =>
                            <button className= {visibilityFilter}
                                key={visibilityFilter} 
                                onClick={() => this.changeVisibilityFilter(visibilityFilter) }>
                                {visibilityFilter.replace('_', ' ')}
                            </button>
                        )
                } </div>
            </div>
        );
    }
}