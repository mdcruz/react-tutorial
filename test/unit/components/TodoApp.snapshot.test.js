import React from 'react';
import TodoApp from 'TodoApp';
import TodoActions from 'TodoActions';
import VisibleTodoList from 'VisibleTodoList';
import SingleTodo from 'SingleTodo';
import renderer from 'react-test-renderer';

describe('TodoApp component', () => {
    it('should render correctly', () => {
        const todoActions = new TodoActions();
        const rendered = renderer.create(<TodoApp todoActions={todoActions} />);
    
        expect(rendered.toJSON()).toMatchSnapshot();
    })
})

describe('SingleTodo component', () => {
    it('should render correctly', () => {
        const rendered = renderer.create(<SingleTodo />);
        expect(rendered.toJSON()).toMatchSnapshot();
    })
})

describe('VisibleTodoList component', () => {
    it('should render correctly', () => {
        const todoActions = new TodoActions();
        const rendered = renderer.create(<VisibleTodoList 
            visibleTodos = {todoActions.getAllTodos()}
            visibilityFilter = "test filter" />);
    
        expect(rendered.toJSON()).toMatchSnapshot();
    })
})