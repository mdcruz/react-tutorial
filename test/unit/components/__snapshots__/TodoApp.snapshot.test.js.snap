// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTodo component should render correctly 1`] = `
<li
  className=""
>
  <input
    checked={undefined}
    onChange={[Function]}
    type="checkbox"
  />
  <label />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;

exports[`TodoApp component should render correctly 1`] = `
<div>
  <h2
    className="appTitle"
  >
    Minimal Todo App built with React
  </h2>
  <form
    className="addTodoForm"
    onSubmit={[Function]}
  >
    <input
      className="todoField"
      onChange={[Function]}
      placeholder="What do you want to do today?"
      type="text"
      value=""
    />
    <button
      disabled={true}
    >
      Add todo
    </button>
  </form>
  <div>
    <h3>
      ALL TODOS
    </h3>
    <ul>
      <li>
        Nothing to see here
      </li>
    </ul>
  </div>
  <div>
     
    <button
      className="ALL_TODOS"
      onClick={[Function]}
    >
      ALL TODOS
    </button>
    <button
      className="ACTIVE_TODOS"
      onClick={[Function]}
    >
      ACTIVE TODOS
    </button>
    <button
      className="COMPLETED_TODOS"
      onClick={[Function]}
    >
      COMPLETED TODOS
    </button>
     
  </div>
</div>
`;

exports[`VisibleTodoList component should render correctly 1`] = `
<div>
  <h3>
    test filter
  </h3>
  <ul>
    <li>
      Nothing to see here
    </li>
  </ul>
</div>
`;
